UPGRADING.md

```diff
@@ -29,6 +31,23 @@ class Grape::API::Instance
 end
 ```
 
+##### `name` (and other caveats) of the mounted API
+
+After the patch, the mounted API is no longer a Named class inheriting from `Grape::API`, it is an anonymous class
+which inherit from `Grape::API::Instance`.
+What this means in practice, is:
+- Generally: you can access the named class from the instance calling the getter `base`.
+- In particular: If you need the `name`, you can use `.to_s`, which will fallback on: `base`.`name`
+
+**Deprecated**
+```ruby
+  payload[:endpoint].options[:for].name
+```
+**New**
+```ruby
+  payload[:endpoint].options[:for].to_s # or `.base.name`
```

Comment:
This will be introduced in 2.1, I think for 1.2.0 it should say `base.name` here, but in 1.2.1 we can recommend `to_s`? Or does `to_s` give you the name before 1.2.0?

Reply:
Prior to 1.2.0, `to_s` on the API object returned the class name.